server/src/main/scala/services/crunch/deskrecs/DesksAndWaitsTerminalProvider.scala

```diff
@@ -38,23 +38,29 @@ case class DesksAndWaitsTerminalProvider(slas: Map[Queue, Int],
 
   def desksAndWaits(minuteMillis: NumericRange[MillisSinceEpoch],
                     loadsByQueue: Map[Queue, Seq[Double]],
-                    deskLimitsProvider: TerminalDeskLimitsLike): Map[Queue, (List[Int], List[Int])] = {
+                    deskLimitsProvider: TerminalDeskLimitsLike): Map[Queue, (Iterable[Int], Iterable[Int])] = {
     val queuesToProcess = loadsByQueue.keys.toSet
 
     queuePriority
       .filter(queuesToProcess.contains)
-      .foldLeft(Map[Queue, (List[Int], List[Int])]()) {
+      .foldLeft(Map[Queue, (Iterable[Int], Iterable[Int])]()) {
         case (queueRecsSoFar, queue) =>
           log.info(s"Optimising $queue")
           val queueWork = adjustedWork(queue, loadsByQueue(queue))
           val minDesks = deskLimitsProvider.minDesksForMinutes(minuteMillis, queue).toSeq
-          val queueDeskAllocations = queueRecsSoFar.mapValues(_._1)
+          val queueDeskAllocations = queueRecsSoFar.mapValues(_._1.toList)
```

Comment:
this would be more readable if you used a match to get the desk allocations out of the tuple

Reply:
or at least add a type annotation on the val

Reply:
agreed. i've added a case to show the value being extracted